Working add members page

j3ffml opened the pull request:
Add members page which requires user logged in and has a house.
Sends an email to each address provided with a link to join the house. that link requires authentication, so if they have an account, they will be redirected to join page, otherwise they have to register, log in and click the join link again... that can be streamlined in the future, but this works for now.

j3ffml committed:
Working add members page

Add members page which requires user logged in and has a house.
On post, send email invites with links to join the house.

j3ffml committed:
Merge branch 'master' of github.com:Full-House-UW/fullhouse into topic-add-members

natmote commented on fullhouse/dashboard/views.py:
```diff
@@ -2,13 +2,20 @@
-from models import House
+from forms import (
+    CreateHouseForm,
+    CreateAnnouncementForm
```
Syntax error. At least run your shit before submitting a pull request.

j3ffml replied:
fixed

j3ffml committed:
Fix syntax error

j3ffml committed:
fix style errors

natmote commented on fullhouse/templates/addmembers/invite_email.txt:
```diff
@@ -0,0 +1,7 @@
+{% load i18n %}
+{{ from_username }}{% trans "has invited you to join a house on" %} {{ site.name }}:
```
there is no space between the username and "has"

natmote commented:
So, this looks pretty good, but the links are pretty lacking. For example, there's no way to get to the dashboard. from the add members page, and there's no way to get to the add members page from the dashboard.

The functionality is there, though, which is great. I'll merge it once you fix the spacing error in the email.

natmote commented:
actually, I just tried to accept an invitation, and it said "invitation accept failed." Did you test this functionality?

j3ffml commented:
It will fail if you accept the invite as a user that is already a member. I should change the message in that case, as it's just the general one (also fails for invalid invitation keys). To correctly test it, accept the invite from an account that isn't already a house member.

j3ffml committed:
fix spacing in invite_email.txt and change failed accept message

Change accept error message (displayed for an invalid join link)
to be more informative.

natmote commented:
What I did was log out, then click the invite link. It sent me to the login page, where I clicked register. I activated my new account, then logged in with my new account, and then clicked the link, and it failed.

natmote commented:
with your new changes, following the same steps, I get "invalid invitation link"

edit: I said the wrong word

j3ffml commented:
did you create the new account with the same email as the one you sent the invite to? it also fails if the emails are not the same. i.e. if you sent the invite to nat@natmote.com and you try to join the house while logged in as herp@herpderp.com, it will fail, even if you have the same link.

natmote commented:
ah, that's what it was. I was just bullshitting the email fields. I'll merge it.
